### PR TITLE
Update to php 7.4 and optimize docker setup

### PR DESCRIPTION
**Pull request by marvincaspar**

### CODE REVIEW

File: dockerfiles/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+  --with-fpm-group=www-data \
+  --enable-intl \
+  --enable-opcache \
+  --enable-zip \
+  --enable-calendar
 
 # Install some must-haves
-RUN yum -y install vim wget sendmail
-RUN yum -y install git
+RUN apt-get install vim wget sendmail
```
Comment (r-brown) [inline dockerfiles/Dockerfile]: Docker fails at this stage with the message
```
Step 3/21 : RUN apt-get install vim wget sendmail
 ---> Running in 7250fefe7bd2
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package vim
E: Unable to locate package wget
E: Unable to locate package sendmail
```

**Review by r-brown — CHANGES_REQUESTED** [review]

File: dockerfiles/Dockerfile (hunk 1 above)
Comment (marvincaspar) [reply dockerfiles/Dockerfile]: Sorry, I will fix it

File: dockerfiles/Dockerfile [hunk 2]
```diff
@@ -48,16 +67,12 @@ COPY config/php/xdebug.ini /usr/local/etc/php/conf.d/xdebug.ini
 # Install Composer
 RUN curl -sS https://getcomposer.org/installer | php
 RUN mv composer.phar /usr/bin/composer
-RUN composer --version
 
 # Install Node.js
-RUN yum -y install nodejs
-RUN yum list installed nodejs
-RUN node -v
```
Comment (r-brown) [inline dockerfiles/Dockerfile:53]: execution below commands is recommended, as these will verify installed versions of needed modules
node -v
npm -v

File: dockerfiles/Dockerfile [hunk 3]
```diff
@@ -48,16 +67,12 @@ COPY config/php/xdebug.ini /usr/local/etc/php/conf.d/xdebug.ini
 # Install Composer
 RUN curl -sS https://getcomposer.org/installer | php
 RUN mv composer.phar /usr/bin/composer
-RUN composer --version
```
Comment (r-brown) [inline dockerfiles/Dockerfile:48]: suggest to keep this command

**Review by r-brown — CHANGES_REQUESTED** [review]
while running build script, the following errors are shown (https://github.com/Labs64/laravel-boilerplate/pulls#run-1)
```
./dockerfiles/bin/prj-build.sh: line 15: npm: command not found
./dockerfiles/bin/prj-build.sh: line 16: npm: command not found
```

**Comment by r-brown** [comment]
After applying new docker configuration and build project (composer, php, **npm**) the project cannot start due to the missing npm artefacts.
see more at https://github.com/Labs64/laravel-boilerplate#run-1 (4. Build project)

**Comment by marvincaspar** [comment]
The link https://github.com/Labs64/laravel-boilerplate/pulls#run-1 is not working for me

**Comment by r-brown** [comment]
> The link https://github.com/Labs64/laravel-boilerplate/pulls#run-1 is not working for me

link fixed https://github.com/Labs64/laravel-boilerplate#run-1 (_4. Build project_)

**Comment by marvincaspar** [comment]
npm should work now

**Comment by r-brown** [comment]
Test
- [x] docker image build
- [x] composer & php dependencies
- [x] nodeJS dependencies
- [x] database setup

**Comment by r-brown** [comment]
Hi @marvincaspar - all works!
Thank you so much for your support!!!

**Pull request merged**